timesketch/api/v1/resources.py

```diff
@@ -861,3 +1023,30 @@ def post(self, sketch_id, story_id):
             db_session.commit()
             return self.to_json(story, status_code=HTTP_STATUS_CODE_CREATED)
         return abort(HTTP_STATUS_CODE_BAD_REQUEST)
+
+
+class QueryResource(ResourceMixin, Resource):
+    """Resource to get a query."""
+    @login_required
+    def post(self, sketch_id):
+        """Handles GET request to the resource.
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
+            story_id: Integer primary key for a story database model
+
+        Returns:
+            A story in JSON (instance of flask.wrappers.Response)
+        """
+        form = ExploreForm.build(request)
+        if form.validate_on_submit():
```

Comment:
I agree (I think..) This requires a big refactor though (all API endpoints and views needs to be changed to be consistent, not just here). I'm gonna follow up on it in a separate CL.